cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -3587,10 +3587,14 @@ void AddBinaryJoin(FunctionRegistry* registry) {
         "binary_join_element_wise", Arity::VarArgs(/*min_args=*/1),
         &binary_join_element_wise_doc, &kDefaultJoinOptions);
     for (const auto& ty : BaseBinaryTypes()) {
-      DCHECK_OK(
-          func->AddKernel({InputType(ty)}, ty,
+      ScalarKernel kernel{KernelSignature::Make({InputType(ty)}, ty, /*is_varargs=*/true),
                           GenerateTypeAgnosticVarBinaryBase<BinaryJoinElementWise>(ty),
-                          BinaryJoinElementWiseState::Init));
+                          BinaryJoinElementWiseState::Init};
+      // This is redundant but expression simplification uses this to potentially replace
+      // calls with null
```

Comment:
```suggestion
```

Reply:
Sorry, my bad - let's not leave this misleading comment here

Reply:
Done